rest/src/test/scala/io/udash/rest/RestApiTest.scala

```diff
@@ -1,19 +1,30 @@
 package io.udash
 package rest
 
-import com.avsystem.commons._
+import com.avsystem.commons.*
+import com.avsystem.commons.misc.ScalaDurationExtensions.durationIntOps
 import io.udash.rest.raw.RawRest
 import io.udash.rest.raw.RawRest.HandleRequest
+import monix.eval.Task
 import monix.execution.Scheduler
 import org.scalactic.source.Position
 import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.time.{Millis, Seconds, Span}
+
+import scala.concurrent.duration.FiniteDuration
 
 abstract class RestApiTest extends AnyFunSuite with ScalaFutures {
+  protected final val MaxConnections: Int = 1 // to timeout quickly
+  protected final val Connections: Int = 10 // > MaxConnections
+  protected final val CallTimeout: FiniteDuration = 300.millis // << idle timeout
+
   implicit def scheduler: Scheduler = Scheduler.global
 
+  private val impl: RestTestApi = RestTestApi.impl()
```

Comment:
I'd consider switching to `def` here to avoid resetting the counter altogether.

Reply:
I am not sure if it will work. We have to persist the instance between calls and checks

Reply:
I thought about fixture level persistence

Reply:
I tried, but macro-related shit beat me, so I put `resetCounter` in `beforeEach`